src/js/models/LottosModel.js

```diff
@@ -17,4 +31,71 @@ export default class LottosModel {
   get list() {
     return this.#lottos.map((value) => value.pickedNumber.join(', '));
   }
+
+  get winningNumberList() {
+    return this.#winningNumberList;
+  }
+
+  set winningNumberList(numberList) {
+    const arrayToNumberList = numberList.map((value) => Number(value));
+    const { LOTTO_NUMBER_LENGTH, BONUS_NUMBER_LENGTH } = LOTTO_SETTING;
+
+    this.#winningNumberList = arrayToNumberList.slice(0, LOTTO_NUMBER_LENGTH);
+    this.#bonusNumber = arrayToNumberList
+      .slice(LOTTO_NUMBER_LENGTH, LOTTO_NUMBER_LENGTH + BONUS_NUMBER_LENGTH)
+      .shift();
+  }
+
+  #isBonusWinner(lottoNumberList, matchCount) {
+    const bonusWinnerRange = LOTTO_SETTING.LOTTO_NUMBER_LENGTH - LOTTO_SETTING.BONUS_NUMBER_LENGTH;
+    return matchCount === bonusWinnerRange && lottoNumberList.includes(this.#bonusNumber);
+  }
+
+  getWinningRank(lottoNumberList) {
+    const { LOTTO_NUMBER_LENGTH, BONUS_NUMBER_LENGTH } = LOTTO_SETTING;
+
+    const matchCount = getListDuplicateCount(lottoNumberList, this.#winningNumberList);
+    const rankNumber = LOTTO_NUMBER_LENGTH + BONUS_NUMBER_LENGTH - matchCount;
+
+    if (this.#isBonusWinner(lottoNumberList, matchCount)) {
+      return LOTTO_SETTING.BONUS_WINNER_INDEX_NUMBER;
+    }
+
+    if (LOTTO_SETTING.LOTTO_NUMBER_LENGTH === matchCount) {
+      return LOTTO_SETTING.WINNER_INDEX_NUMBER;
+    }
+
+    return rankNumber;
+  }
+
+  getWinningCount() {
+    const output = Array.from({ length: LOTTO_SETTING.RACKING_START_NUMBER }, () => 0);
```

Comment:
이건 fill 을 이용해 초기화할 수도 있을 거 같습니다! (수정할 필요는 없어용)
```suggestion
    const output = Array(LOTTO_SETTING.RACKING_START_NUMBER).fill(0)
```

Reply:
헉..
일단 추후 코드 보며, 다시 리마인드하기 위해 수정했습니다!